AK/Time.h

```diff
@@ -66,6 +72,55 @@ inline int years_to_days_since_epoch(int year)
     return days;
 }
 
+/*
+ * Represents a time amount in a "safe" way.
+ * Minimum: 0 seconds, 0 nanoseconds
+ * Maximum: 2**63-1 seconds, 999'999'999 nanoseconds
+ * If any operation (e.g. 'from_timeval' or operator-) would over- or underflow, the closest legal value is returned instead.
+ * Inputs (e.g. to 'from_timespec') are allowed to be in non-normal form (e.g. "1 second, 2'012'345'678 nanoseconds" or "1 second, -2 microseconds").
+ * Outputs (e.g. from 'to_timeval') are always in normal form.
+ */
+class Time {
+public:
+    Time() = default;
+    Time(const Time&) = default;
+
+    static Time from_seconds(i64 seconds) { return Time(seconds, 0); }
+    static Time from_nanoseconds(i32 nanoseconds);
+    static Time from_timespec(const struct timespec&);
+    static Time from_timeval(const struct timeval&);
+    static Time min() { return Time(-0x8000'0000'0000'0000LL, 0); };
+    static Time zero() { return Time(0, 0); };
+    static Time max() { return Time(0x7fff'ffff'ffff'ffffLL, 999'999'999); };
+
+    // Truncates "2.8 seconds" to 2 seconds.
+    // Truncates "-2.8 seconds" to -2 seconds.
+    i64 to_truncated_seconds() const;
+    timespec to_timespec() const;
+    timeval to_timeval() const;
+
+    bool operator==(const Time& other) const { return this->m_seconds == other.m_seconds && this->m_nanoseconds == other.m_nanoseconds; }
+    bool operator!=(const Time& other) const { return !(*this == other); }
+    Time operator+(const Time& other) const;
+    Time operator-(const Time& other) const;
+    bool operator<(const Time& other) const;
+    bool operator<=(const Time& other) const;
+    bool operator>(const Time& other) const;
+    bool operator>=(const Time& other) const;
```

Comment:
I already looked into this particular method for `DistinctNumeric`, and we can't easily support it because it is required to return a particular type in `std::…`, which we don't have.